src/index.js

```diff
@@ -104,12 +105,18 @@ export default {
     );
     return snapshotLayout;
   },
-  getContextMenu (obj, menu) {
+  async getContextMenu (obj, menu) {
     if (!this.$scope.layout.allowexportxls) {
       return menu;
     }
 
-    if (this.backendApi.model.layout.qMeta.privileges.indexOf('exportdata') !== -1) {
+    // Export as XLS is removed from desktop because the desktop wrapper blocks downloads.
+    // isPersonalMode returns true for both desktop and QCS
+    // By checking both if has download dialog and if is QCS can enable Export as XLS option on QCS
+    const app = qlik.currApp(this);
+    const isPersonalResult = await app.global.isPersonalMode();
+
+    if (this.backendApi.model.layout.qMeta.privileges.indexOf('exportdata') !== -1 || (flags.isEnabled('DOWNLOAD_USE_REPORTING') && isPersonalResult && isPersonalResult.qReturn)) {
```

Comment:
I don't think this correct for the desktop case
as if privileges contains 'exportdata' in desktop it will show the menu and as from waht I understand it should not.